### ISSUE TITLE
DocFX failing to process azure-servicebus Python package API content

### ISSUE THREAD

**Issue by dend** [dotnet/docfx#4689]
I am running a standalone version of DocFX, and trying to document the [`azure-servicebus` ](https://pypi.org/project/azure-servicebus/) Python package. The documentation for the package is generated via [`sphinx-docfx-yaml`](https://github.com/docascode/sphinx-docfx-yaml).

The documentation that is generated via the tool is attached to this issue:
[Archive.zip](https://github.com/dotnet/docfx/files/3245351/Archive.zip)

The build with the generated YAML files is failing, however, with the following error message:

```bash
Build failed.
[19-06-02 11:28:21.159]Warning:[MetadataCommand.ExtractMetadata]No project detected for extracting metadata.
[19-06-02 11:28:21.159]Error:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.Apply Templates](api/servicebus.receive_handler.SessionReceiver.yml)Error transforming model generated from "api/servicebus.receive_handler.SessionReceiver.yml" using "UniversalReference.html.primary.js". To get the detailed raw model, please run docfx with debug mode --debug. Error running Transform function inside template preprocessor:  
[19-06-02 11:28:21.159]Error:Error transforming model generated from "api/servicebus.receive_handler.SessionReceiver.yml" using "UniversalReference.html.primary.js". To get the detailed raw model, please run docfx with debug mode --debug. Error running Transform function inside template preprocessor:  
     1 Warning(s)
     2 Error(s)
```

**Comment by superyyrrzz** [comment]
This is an issue related to template forking. Docfx template expect the YAML files be self contained. In `servicebug.receive_handler.SessionReceiver.yml`, it has the following section:
```yml
  syntax:
    content: set_session_state(state)
    parameters:
    - description: The state value.
      id: state
      type:
      - str, bytes
      - bytearray
```
Docfx template expect the YAML's `references` section to have a definition of `bytearray`, but it has not.

I see another issue about `str, bytes` here. Docfx template will not split this into 2 type, while docs.msft template does this trick.

It seems the `sphinx-docfx-yaml` tool is coupled with docs.msft template now. The temporary solution of this bug is to sync docs.msft template into docfx template. In long terms, we need to define new model in SDP, and ensure there is only one template since docfx v3.

**Comment by dend** [comment]
@superyyrrzz just to check in - is this something that the team will investigate, or is this a `wont-fix`? This issue is now open for more than a year 😀

**Comment by superyyrrzz** [comment]
It may be fixed by #5781. You can have a try.

Per [current strategy](https://github.com/dotnet/docfx/blob/dev/Roadmap.md#v2-strategy), team will not invest many resources on it. However, we are happy to accept PR from community.
